tests/components/element-editor/basic-details/basic-details.spec.tsx

```diff
@@ -108,27 +100,117 @@ describe("BasicDetails", () => {
+                "dropdown-item-value"
+            );
+
+            expect(dropdownItemLabels[0]).toBeInTheDocument();
+            expect(dropdownItemValues[0]).toBeInTheDocument();
+
+            expect(
+                screen.queryByText("Pre-selected value (optional)")
+            ).not.toBeInTheDocument();
+
+            fireEvent.change(dropdownItemLabels[0], {
+                target: { value: "New Label" },
+            });
+            fireEvent.change(dropdownItemValues[0], {
+                target: { value: "New Value" },
+            });
+
+            expect(dropdownItemLabels[0]).toHaveValue("New Label");
+            expect(dropdownItemValues[0]).toHaveValue("New Value");
+
+            const preselectedValue = await screen.findByText(
+                "Pre-selected value (optional)"
+            );
+
+            expect(preselectedValue).toBeInTheDocument();
+        });
+    });
+
+    describe("submitting the form for dropdown element", () => {
+        it("should not be able to submit the form if there are not at least 2 valid dropdown items", async () => {
+            renderComponent(EElementType.DROPDOWN, {
+                builderContext: {
+                    focusedElement: MOCK_FOCUSED_DROPDOWN_ELEMENT,
+                    elements: MOCK_ELEMENTS,
+                },
+            });
+
+            const submitButton = screen.getByText("Submit");
+            fireEvent.click(submitButton);
+
+            const labelError = await screen.findAllByText(
+                "Option label required."
+            );
+            const valueError = await screen.findAllByText(
+                "Option value required."
+            );
+
+            // Initially there are 2 empty dropdown items
+            expect(labelError).toHaveLength(2);
+            expect(valueError).toHaveLength(2);
+        });
+    });
 });
 
+// =============================================================================
+// INTERFACES
+// =============================================================================
+
+interface IProps {
+    elementType: EElementType;
+}
+
 // =============================================================================
 // HELPER FUNCTIONS
 // =============================================================================
 
-const MyTestComponent = () => {
+const MyTestComponent = ({ elementType }: IProps) => {
     const methods = useForm({
         mode: "onTouched",
-        resolver: yupResolver(SchemaHelper.buildSchema(EElementType.EMAIL)),
+        resolver: yupResolver(SchemaHelper.buildSchema(elementType)),
+        defaultValues: {
+            type: elementType,
+        },
     });
+    const onSubmit = jest.fn;
     return (
         <FormProvider {...methods}>
-            <BasicDetails />
+            <form onSubmit={methods.handleSubmit(onSubmit)}>
+                <BasicDetails />
+                <button type="submit">Submit</button>
+            </form>
         </FormProvider>
     );
 };
```

Comment:
I notice many of our tests require `FormProvider`, may I request to update `TestHelper.withProviders()` to include the `FormProvider`?

`RenderOptions` can be updated to support various customisation options for the `FormProvider`